Code/Tools/Standalone/Source/Editor/LuaEditor.cpp

```diff
@@ -74,6 +74,10 @@ int main(int argc, char* argv[])
         // if its in GUI mode or not.
     }
 
+    if (AZ::AllocatorInstance<AZ::SystemAllocator>::IsReady())
+    {
+        AZ::AllocatorInstance<AZ::SystemAllocator>::Destroy();
+    }
```

Comment:
The Irony of this is we can also solve this problem by removing code as well

If we remove the following [block](https://github.com/o3de/o3de/pull/4130/files/1438c604190c90ac040583e73bb8d36266939be5..e96b75127882bb7cb4305761af1c1db2f58f425d#diff-6f8b9c30a9a12c11af8d1ccf1a30fd2c7bb8ed7b310e4c426eb4c7bc7d3c843aR34-R44) of lines above, then the ComponentApplication takes care of creating and destroying the Allocator

Reply:
I think it is done before the component application because it wants to FileIOInstance early before the ComponentApplication does

Reply:
That is not needed anymore as the `LUAEditor::Application` inherits from `AzFramework::Applicatoin` which creates a FileIOInstance in it's constructor, so the explicit creation of the FileIOInstance here still isn't needed.

Reply:
Well it looks like `LuaEditor::Application` does not inherit from `AzFramework::Application`.

It goes [LuaEditor::Application](https://github.com/o3de/o3de/blob/development/Code/Tools/Standalone/Source/LuaIDEApplication.h#L13) ->[StandaloneTools::BaseApplication](https://github.com/o3de/o3de/blob/development/Code/Tools/Standalone/Source/StandaloneToolsApplication.h#L20) -> [LegacyFramework::Application](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/UI/LegacyFramework/Core/EditorFrameworkApplication.h#L48) -> [AZ::ComponentApplication](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/ComponentApplication.h#L77)

So carry on.

Reply:
Dang